lib/cadet_web/controllers/missions_controller.ex

```diff
@@ -0,0 +1,190 @@
+defmodule CadetWeb.MissionsController do
+  use CadetWeb, :controller
+
+  use PhoenixSwagger
+
+  swagger_path :index do
+    get("/missions")
+
+    summary("Get a list of all missions")
+
+    security([%{JWT: []}])
+
+    produces("application/json")
+
+    response(200, "OK", Schema.ref(:MissionsList))
+    response(401, "Unauthorised")
+  end
+
+  swagger_path :open do
+    get("/missions/open")
+
+    summary("Get a list of open missions")
+
+    security([%{JWT: []}])
+
+    produces("application/json")
+
+    response(200, "OK", Schema.ref(:MissionsList))
+    response(401, "Unauthorised")
+  end
+
+  swagger_path :show do
+    get("/missions/{missionId}")
+
+    summary("Get information about one particular mission.")
+
+    security([%{JWT: []}])
+
+    produces("application/json")
+
+    parameters do
+      missionId(:path, :integer, "mission id", required: true)
+    end
+
+    response(200, "OK", Schema.ref(:Mission))
+    response(400, "Missing parameter(s) or invalid missionId")
+    response(401, "Unauthorised")
+  end
+
+  swagger_path :questions do
+    get("/missions/{missionId}/questions")
+
+    summary("Get questions contained inside a mission. Response is either `mcq` or `programming`")
```

Comment:
Does this mean that the response can contain (exclusive) either of the two? Right now the structure appears to be two arrays, one for `programming` and the other for `mcq`

Reply:
Yup, it's exclusive. Cos the swagger library we're using is still on OpenAPI 2.0 which doesn't allow `oneOf` 😞